operators/src/processing/column_range_filter.rs

```diff
@@ -0,0 +1,283 @@
+use crate::adapters::FeatureCollectionChunkMerger;
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+    Operator, QueryContext, QueryProcessor, QueryRectangle, TypedVectorQueryProcessor,
+    VectorOperator, VectorQueryProcessor, VectorResultDescriptor,
+};
+use crate::error;
+use crate::util::input::StringOrNumber;
+use crate::util::Result;
+use failure::_core::marker::PhantomData;
+use failure::_core::ops::RangeInclusive;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use geoengine_datatypes::collections::FeatureCollection;
+use geoengine_datatypes::primitives::{FeatureDataType, FeatureDataValue, Geometry};
+use geoengine_datatypes::util::arrow::ArrowTyped;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+use std::convert::TryInto;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ColumnRangeFilterParams {
+    pub column: String,
+    pub ranges: Vec<(StringOrNumber, StringOrNumber)>,
```

Comment:
this allows mixed ranges like (0, "hi"). It only fails when the query processor is executed. I'd prefer it to fail much earlier. So either a Vec<StringOrNumberRange> that ensures validity, or maybe even two semantically separated operators NumericRangeFilter, TextualRangeFilter that are implemented using a generic ColumnRangeFilter with a parametrized typed range handler.

Reply:
On the other hand, you want mixed ranges like `(0, 1.5)`.

Is there an advantage for separate operators? I mean, either we can check something prior to processing or we can't.

Reply:
why do we want mixed ranges? just state `(0.0, 1.5)`.

Advantage of separate opterators:
- more visible distinction of the operators
- more easily extensible for data type specific additional parameters. e.g. ignore case for text

but I'm not hard set on separation.

At least we should check the validity of the input on parsing however, s.t. we don't end up with workflows in the registry that will always fail when trying to create a query processor from them

Reply:
47f20b4